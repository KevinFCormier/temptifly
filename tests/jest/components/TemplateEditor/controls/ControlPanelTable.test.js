'use strict'

import React from 'react'
import ControlPanelTable from '../../../../../src/controls/ControlPanelTable'
import renderer from 'react-test-renderer'

import { tableControl } from '../../TestingData'

describe('ControlPanelTable component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()

    const component = renderer.create(
      <ControlPanelTable
        key={'key'}
        control={tableControl}
        fetchData={fn}
        controlId={'controlId'}
        handleChange={fn}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
