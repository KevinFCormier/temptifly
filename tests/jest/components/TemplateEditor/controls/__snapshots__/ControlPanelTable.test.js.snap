// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanelTable component renders as expected 1`] = `
<div
  className="creation-view-controls-table-container"
>
  <div
    className="creation-view-controls-table"
  >
    <div
      className="bx--data-table-v2-container"
    >
      <section
        aria-label={undefined}
        className="bx--table-toolbar"
      >
        <div
          className="bx--toolbar-search-container"
        >
          <div
            aria-labelledby="resource-search-bar-label"
            className="bx--search bx--search--sm bx--search--light"
            role="search"
          >
            <svg
              alt="Provide a description that will be used as the title"
              aria-label="Provide a description that will be used as the title"
              className="bx--search-magnifier"
              fill={undefined}
              fillRule="evenodd"
              height="16"
              name={undefined}
              role="img"
              style={undefined}
              viewBox="0 0 16 16"
              width="16"
            >
              <title>
                Provide a description that will be used as the title
              </title>
              <path
                d="M6.5 12a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11zm4.936-1.27l4.563 4.557-.707.708-4.563-4.558a6.5 6.5 0 1 1 .707-.707z"
              />
            </svg>
            <label
              className="bx--label"
              htmlFor="resource-search-bar"
              id="resource-search-bar-label"
            />
            <input
              className="bx--search-input"
              id="resource-search-bar"
              onChange={[Function]}
              placeholder=""
              type="text"
              value=""
            />
            <button
              aria-label={undefined}
              className="bx--search-close bx--search-close--hidden"
              onClick={[Function]}
              type="button"
            >
              <svg
                alt="Provide a description that will be used as the title"
                aria-label="Provide a description that will be used as the title"
                className={undefined}
                fill={undefined}
                fillRule="evenodd"
                height="16"
                name={undefined}
                role="img"
                style={undefined}
                viewBox="0 0 16 16"
                width="16"
              >
                <title>
                  Provide a description that will be used as the title
                </title>
                <path
                  d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414 8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          className="bx--toolbar-content"
        />
      </section>
      <table
        className="bx--data-table-v2 resource-table"
      >
        <thead
          className={undefined}
        >
          <tr
            className={undefined}
          >
            <th
              scope="col"
            >
              <button
                className="bx--table-sort-v2 bx--table-sort-v2--ascending"
                data-key="hostName"
                onClick={[Function]}
                title={undefined}
              >
                <span
                  className="bx--table-header-label"
                >
                  creation.ocp.host.name
                </span>
                <svg
                  alt="Provide a description that will be used as the title"
                  aria-label="Provide a description that will be used as the title"
                  className="bx--table-sort-v2__icon"
                  fill={undefined}
                  fillRule="evenodd"
                  height="5"
                  name="caret--down"
                  role="img"
                  style={undefined}
                  viewBox="0 0 10 5"
                  width="10"
                >
                  <title>
                    Provide a description that will be used as the title
                  </title>
                  <path
                    d="M0 0l5 4.998L10 0z"
                  />
                </svg>
              </button>
            </th>
            <th
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          className={undefined}
        />
      </table>
    </div>
  </div>
  <div
    className="bx--pagination"
    disabled={true}
  >
    <div
      className="bx--pagination__left"
    >
      <span
        className="bx--pagination__text"
      >
        Items per page:
      </span>
      <div
        className="bx--form-item"
      >
        <div
          className="bx--select bx--select--inline"
        >
          <label
            className="bx--label bx--visually-hidden"
            htmlFor="bx-pagination-select-resource-table-pagination"
          >
            Items per page:
          </label>
          <select
            aria-invalid={undefined}
            className="bx--select-input"
            data-invalid={undefined}
            disabled={undefined}
            id="bx-pagination-select-resource-table-pagination"
            onChange={[Function]}
            value={5}
          >
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={5}
            >
              5
            </option>
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={10}
            >
              10
            </option>
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={20}
            >
              20
            </option>
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={50}
            >
              50
            </option>
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={75}
            >
              75
            </option>
            <option
              className="bx--select-option"
              disabled={false}
              hidden={false}
              value={100}
            >
              100
            </option>
          </select>
          <svg
            alt="open list of options"
            aria-label="open list of options"
            className="bx--select__arrow"
            fill={undefined}
            fillRule="evenodd"
            height="5"
            name={undefined}
            role="img"
            style={undefined}
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              open list of options
            </title>
            <path
              d="M0 0l5 4.998L10 0z"
            />
          </svg>
        </div>
      </div>
      <span
        className="bx--pagination__text"
      >
         |  
        undefined undefined
      </span>
    </div>
    <div
      className="bx--pagination__right bx--pagination--inline"
    >
      <button
        aria-label="Previous page"
        className="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index"
        disabled={true}
        onClick={[Function]}
      >
        <svg
          alt="Previous page"
          aria-label="Previous page"
          className="bx--pagination__button-icon"
          fill={undefined}
          fillRule="evenodd"
          height="12"
          name={undefined}
          role="img"
          style={undefined}
          viewBox="0 0 7 12"
          width="7"
        >
          <title>
            Previous page
          </title>
          <path
            d="M1.45 6.002L7 11.27l-.685.726L0 6.003 6.315 0 7 .726z"
          />
        </svg>
      </button>
      <button
        aria-label="Next page"
        className="bx--pagination__button bx--pagination__button--forward"
        disabled={true}
        onClick={[Function]}
      >
        <svg
          alt="Next page"
          aria-label="Next page"
          className="bx--pagination__button-icon"
          fill={undefined}
          fillRule="evenodd"
          height="12"
          name={undefined}
          role="img"
          style={undefined}
          viewBox="0 0 7 12"
          width="7"
        >
          <title>
            Next page
          </title>
          <path
            d="M5.569 5.994L0 .726.687 0l6.336 5.994-6.335 6.002L0 11.27z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
